### ISSUE TITLE
dm chaos test failed because of `Unsupported collation when new collation is enabled: 'latin1_swedish_ci'`

### ISSUE THREAD

**Issue by amyangfei** [pingcap/tiflow#7592]
### What did you do?

run chaos test in dm on dataflow engine.
ref: https://github.com/pingcap/tiflow/actions/runs/3452878296

This can be reproduced with mariaDB

```bash
mysql> select version();
+-------------------------------------------+
| version()                                 |
+-------------------------------------------+
| 10.9.4-MariaDB-1:10.9.4+maria~ubu2204-log |
+-------------------------------------------+
1 row in set (0.00 sec)

mysql> create database hello charset latin1;
Query OK, 1 row affected (0.00 sec)

mysql> use hello;
Database changed
mysql> create table t1 (id int primary key auto_increment);
Query OK, 0 rows affected (0.01 sec)

mysql> show create table hello.t1;
+-------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Table | Create Table                                                                                                                                                       |
+-------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| t1    | CREATE TABLE `t1` (
  `id` int(11) NOT NULL AUTO_INCREMENT,
  PRIMARY KEY (`id`)
) ENGINE=InnoDB AUTO_INCREMENT=4 DEFAULT CHARSET=latin1 COLLATE=latin1_swedish_ci |
+-------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.01 sec)
```

- Use dumpling to export database `hello`, observe table schema file

```sql
> cat export4/hello.t1-schema.sql
/*!40101 SET NAMES binary*/;
CREATE TABLE `t1` (
  `id` int(11) NOT NULL AUTO_INCREMENT,
  PRIMARY KEY (`id`)
) ENGINE=InnoDB AUTO_INCREMENT=4 DEFAULT CHARSET=latin1 COLLATE=latin1_swedish_ci;
```

### What did you see instead?

```go
[2022/11/12 23:13:47.166 +00:00] [ERROR] [restore.go:686] ["execute SQL: CREATE TABLE IF NOT EXISTS `dmjob`.`tb0` (`id` INT(11) DEFAULT NULL,`name` VARCHAR(255) DEFAULT NULL,`age` INT(11) DEFAULT NULL,`team_id` INT(11) DEFAULT NULL,UNIQUE `id`(`id`)) ENGINE = InnoDB DEFAULT CHARACTER SET = LATIN1 DEFAULT COLLATE = LATIN1_SWEDISH_CI; failed"] [job_id=7d25a033-318f-4884-bc90-e1a8cc005caf] [worker_id=57fa70f3-e3c9-48bc-9bdf-88f6ce08b130] [db=dmjob] [table=tb0] [takeTime=43.770556ms] [error="run create schema job failed: Error 1273: Unsupported collation when new collation is enabled: 'latin1_swedish_ci'"]                                                                                                                                             [2022/11/12 23:13:47.166 +00:00] [ERROR] [restore.go:772] ["restore all schema failed"] [job_id=7d25a033-318f-4884-bc90-e1a8cc005caf] [worker_id=57fa70f3-e3c9-48bc-9bdf-88f6ce08b130] [takeTime=140.413424ms] [error="[Lightning:Restore:ErrCreateSchema]create schema failed, table: `dmjob`.`tb0`, stmt: restore table schema: run create schema job failed: Error 1273: Unsupported collation when new collation is enabled: 'latin1_swedish_ci'"]
[2022/11/12 23:13:47.166 +00:00] [ERROR] [restore.go:466] ["run failed"] [job_id=7d25a033-318f-4884-bc90-e1a8cc005caf] [worker_id=57fa70f3-e3c9-48bc-9bdf-88f6ce08b130] [step=1] [error="[Lightning:Restore:ErrCreateSchema]create schema failed, table: `dmjob`.`tb0`, stmt: restore table schema: run create schema job failed: Error 1273: Unsupported collation when new collation is enabled: 'latin1_swedish_ci'"]
[2022/11/12 23:13:47.166 +00:00] [ERROR] [restore.go:476] ["the whole procedure failed"] [job_id=7d25a033-318f-4884-bc90-e1a8cc005caf] [worker_id=57fa70f3-e3c9-48bc-9bdf-88f6ce08b130] [takeTime=140.477429ms] [error="[Lightning:Restore:ErrCreateSchema]create schema failed, table: `dmjob`.`tb0`, stmt: restore table schema: run create schema job failed: Error 1273: Unsupported collation when new collation is enabled: 'latin1_swedish_ci'"]
[2022/11/12 23:13:47.166 +00:00] [ERROR] [lightning.go:345] ["failed to runlightning"] [job_id=7d25a033-318f-4884-bc90-e1a8cc005caf] [worker_id=57fa70f3-e3c9-48bc-9bdf-88f6ce08b130] [task=7d25a033-318f-4884-bc90-e1a8cc005caf] [unit=lightning-load] [error="[code=34019:class=load-unit:scope=not-set:level=high], RawCause: [Lightning:Restore:ErrCreateSchema]create schema failed, table: `dmjob`.`tb0`, stmt: restore table schema: run create schema job failed: Error 1273: Unsupported collation when new collation is enabled: 'latin1_swedish_ci'"] [errorVerbose="[code=34019:class=load-unit:scope=not-set:level=high], RawCause: [Lightning:Restore:ErrCreateSchema]create schema failed, table: `dmjob`.`tb0`, stmt: restore table schema: run create schema job failed: Error 1273: Unsupported collation when new collation is enabled: 'latin1_swedish_ci'\ngithub.com/pingcap/tiflow/dm/pkg/terror.(*Error).Delegate\n\tgithub.com/pingcap/tiflow/dm/pkg/terror/terror.go:322\ngithub.com/pingcap/tiflow/dm/loader.(*LightningLoader).runLightning\n\tgithub.com/pingcap/tiflow/dm/loader/lightning.go:275\ngithub.com/pingcap/tiflow/dm/loader.(*LightningLoader).restore\n\tgithub.com/pingcap/tiflow/dm/loader/lightning.go:336\ngithub.com/pingcap/tiflow/dm/loader.(*LightningLoader).Process\n\tgithub.com/pingcap/tiflow/dm/loader/lightning.go:390\ngithub.com/pingcap/tiflow/engine/executor/dm.(*unitHolderImpl).Init.func1\n\tgithub.com/pingcap/tiflow/engine/executor/dm/unitholder.go:135\nruntime.goexit\n\truntime/asm_amd64.s:1594"]
```

### Versions of the cluster

Dataflow Engine version (run `tiflow version`):

master since https://github.com/pingcap/tiflow/commit/04173f64ea66bd4e709f8274e5bd1ffe60e70625 until now

**Comment by amyangfei** [comment]
I have tested the same SQL on mariadb 10.9.3, the create table SQL doesn't contain collation.

```sql
mysql> create database hello charset latin1;
Query OK, 1 row affected (0.00 sec)

mysql> use hello;
Database changed
mysql> create table t1 (id int primary key auto_increment);
Query OK, 0 rows affected (0.01 sec)

mysql> show create table hello.t1;
+-------+-------------------------------------------------------------------------------------------------------------------------+
| Table | Create Table                                                                                                            |
+-------+-------------------------------------------------------------------------------------------------------------------------+
| t1    | CREATE TABLE `t1` (
  `id` int(11) NOT NULL AUTO_INCREMENT,
  PRIMARY KEY (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=latin1 |
+-------+-------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.01 sec)

mysql> select version();
+-------------------------------------------+
| version()                                 |
+-------------------------------------------+
| 10.9.3-MariaDB-1:10.9.3+maria~ubu2204-log |
+-------------------------------------------+
1 row in set (0.00 sec)
```

**Comment by amyangfei** [comment]
Closed by #7588
